Tasks/JenkinsDownloadArtifacts/task.json

```diff
@@ -122,6 +127,67 @@
             "visibleRule": "downloadCommitsAndWorkItems == true",
             "groupName": "advanced",
             "helpMarkDown": "Optional file name suffix for commits and workitem attachment. Attachment will be created with commits_{suffix}.json and workitem_{suffix}.json. If this input is not provided attachments will be create with the name commits.json and workitems.json"
+        },
+        {
+            "name": "propagatedArtifacts",
+            "type": "boolean",
+            "label": "Artifacts are propagated to Azure",
+            "defaultValue": "false",
+            "required": false,
+            "groupName": "propagatedArtifactsGroup",
+            "helpMarkDown": "Check this if Jenkins artifacts are propagated."
+        },
+        {
+            "name": "artifactProvider",
+            "type": "pickList",
+            "label": "Artifact Provider",
+            "defaultValue": "azureStorage",
+            "required": true,
+            "options": {
+                "azureStorage":"Azure Storage"
+            },
+            "groupName": "propagatedArtifactsGroup",
+            "visibleRule": "propagatedArtifacts == true",
+            "helpMarkDown": "Choose propagated artifact provider."
```

Comment:
Be more descriptive and provide relevant links in helpMarkDown. E.g. "Choose the external storage provider Jenkins artifacts were propagated to."